Scripts/WikiUpdater.cs

```diff
@@ -62,93 +132,225 @@ public async Task<bool> Run(CancellationToken cancellationToken)
 
         return true;
     }
 
-    private async Task<IHtmlElement> FetchOrganellesRootPage(CancellationToken cancellationToken)
+    /// <summary>
+    ///   Fetches a page from the online wiki
+    /// </summary>
+    /// <returns>The HTML content of the page</returns>
+    private async Task<IHtmlElement> FetchRootPage(string url, string categoryName, CancellationToken cancellationToken)
     {
-        ColourConsole.WriteInfoLine("Fetching organelles root page");
-        var body = (await HtmlReader.RetrieveHtmlDocument(ORGANELLE_CATEGORY, cancellationToken)).Body!;
-        pageNames.Add("Organelles", "OrganellesRoot");
+        ColourConsole.WriteInfoLine($"Fetching {categoryName.ToLowerInvariant()} root page");
+        var body = (await HtmlReader.RetrieveHtmlDocument(url, cancellationToken)).Body!;
+        pageNames.Add(categoryName, $"{categoryName}Root");
         return body;
     }
 
-    private async Task<List<IHtmlElement>> FetchOrganellePages(CancellationToken cancellationToken)
+    /// <summary>
+    ///   Fetches all the pages from a category, ignores pages in the OnlyOnline category
+    /// </summary>
+    /// <returns>A list of all the pages' HTML content</returns>
+    private async Task<List<IHtmlElement>> FetchPagesFromCategory(string categoryUrl,
+        string pageType, CancellationToken cancellationToken)
     {
-        ColourConsole.WriteInfoLine("Fetching organelle pages");
-        var organellePages = new List<IHtmlElement>();
+        ColourConsole.WriteInfoLine($"Fetching {pageType} pages");
+        var allPages = new List<IHtmlElement>();
 
-        // Get the list of organelles from the category page on the wiki
-        var categoryBody = (await HtmlReader.RetrieveHtmlDocument(ORGANELLE_CATEGORY, cancellationToken)).Body!;
-        var organelles = categoryBody.QuerySelectorAll(".mw-category-group > ul > li");
+        var textInfo = CultureInfo.InvariantCulture.TextInfo;
 
-        foreach (var organelle in organelles)
+        // Get the list of pages from the category page on the wiki
+        var categoryBody = (await HtmlReader.RetrieveHtmlDocument(categoryUrl, cancellationToken)).Body!;
+        var pages = categoryBody.QuerySelectorAll(CATEGORY_PAGES_SELECTOR);
+
+        foreach (var page in pages)
         {
-            var name = organelle.TextContent.Trim();
+            var name = page.TextContent.Trim();
             var url = $"https://wiki.revolutionarygamesstudio.com/wiki/{name.Replace(" ", "_")}";
 
-            ColourConsole.WriteInfoLine($"Found organelle {name}");
+            ColourConsole.WriteInfoLine($"Found {pageType} {name}");
 
             var body = (await HtmlReader.RetrieveHtmlDocument(url, cancellationToken)).Body!;
-            var internalName = body.QuerySelector("#info-box-internal-name")!.TextContent.Trim();
+            var internalName = textInfo.ToTitleCase(name).Replace(" ", string.Empty);
```

Comment:
Can this be moved down and integrated into a ternary with lines 200 to 203? I think it needs to be clearer for future maintenance that `internalName` isn't always the page name.